05week/spaceTravelToMars.js

```diff
@@ -9,7 +9,67 @@ let jobTypes = {
   programmer: 'Any Ship!'
 };
 
-// Your code here
+// create a crew member class
+// constructor should take in: name, job, special skill
+// when a crew member is created, they're not associated with any ship
+// have an attribute "ship", this is the ship they're in
+
+class CrewMember {
+  constructor(name, job, specialSkill) {
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+    this.ship = null;
+  }
+
+  // this method should add THIS crewmember to the ship being passed in
+  // !! an entire ship instance is passed in, not just the name
+  // !! the entire crewmember is added to the ships array of crew
+  
+  enterShip(ship) {
+    if (this.ship === null) {
+      this.ship = ship;
+      ship.crew.push(this);
+    }
+  }
+
+}
+
+// create a ship class
+// 3 attributes: name of ship, type of ship, ability of ship
+// when a ship is created, should have no crew members
+// create a slot for crew, that starts out empty
+// should have a method called missionStatement()
+//    if there is a crew member that can activate it, it should return its ability
+//    
+//    otherwise it should return "can't perform a mission yet"
+
+class Ship{
+  constructor(name, type, ability) {
+    this.name = name;
+    this.type = type;
+    this.ability = ability;
+    this.crew = [];
+  }
+
+  // this method should return the ships ability if there is a crew member
+  // whose job matches up with the ships type
+  // else should return "Can't perform a mission yet"
+  missionStatement() {
+    for (let i=0; i<this.crew.length; i++) {
+      if (this.type == this.crew[i].job) {
+        return `${this.ability}`;
+      } else {
+        return `Can't perform a mission yet.`;
```

Comment:
this will not work if the second crew member matches the ships ability since you are exiting the function after 1 iteration of the loop.
The "can't perform" message should be returned after you have looked at all crew members